packages/visualizations/src/components/utils/ColorsLegend.svelte

```diff
@@ -1,13 +1,47 @@
 <svelte:options immutable={true} />
 
 <script lang="ts">
+    import { onDestroy, onMount } from 'svelte';
+    import { debounce } from 'lodash';
     import type { DataBounds, ColorsScale, LegendVariant } from '../types';
     import { defaultCompactLegendNumberFormat } from './formatter';
+
     // options to customize the component
     export let dataBounds: DataBounds;
     export let colorsScale: ColorsScale;
     export let variant: LegendVariant;
     export let title: string;
+
+    // the part below is related to labels rotation
+    let legendWidth: number;
+    let colorBoxWidth: number;
+    const labelsWidth: number[] = [];
+    let maxLabelsSize: number;
+    let numberOfLabels: number;
+    let displayVertical: boolean;
+    const handleLabelRotation = (): void => {
+        if (colorsScale.type === 'palette' && labelsWidth.length !== 0) {
+            const availableWidthPerLabel: number = legendWidth / numberOfLabels - 3;
+            colorBoxWidth = legendWidth / numberOfLabels;
+            numberOfLabels = colorsScale.type === 'palette' ? colorsScale.colors.length + 1 : 2;
+            maxLabelsSize = labelsWidth.reduce((a, b) => Math.max(a, b));
+            if (availableWidthPerLabel < maxLabelsSize) {
+                displayVertical = true;
+            } else {
+                displayVertical = false;
+            }
+        }
+    };
+    const rotationDebounce = debounce(handleLabelRotation, 200);
+    onMount(() => handleLabelRotation());
+    // Makes rotationDebounce reactive to legendWith and colorScale changes
+    $: if (legendWidth) {
+        rotationDebounce();
+    }
+    $: if (colorsScale) {
+        rotationDebounce();
+    }
+    onDestroy(() => rotationDebounce.cancel);
```

Comment:
I think it should be either `onDestroy(rotationDebounce.cancel);` or `onDestroy(() => rotationDebounce.cancel());`

I guess the first one is nicer, and it could be `onMount(handleLabelRotation);` as well

Reply:
👍